plugins/scenario/World.ts

```diff
@@ -1,52 +1,84 @@
 import { SignerWithAddress } from '@nomiclabs/hardhat-ethers/signers';
-import { HardhatRuntimeEnvironment } from 'hardhat/types';
 
 // NB: this couples this plugin to deployment manager plugin
 import { DeploymentManager } from '../deployment_manager/DeploymentManager';
+import hreForBase from './utils/hreForBase';
+import { impersonateAddress } from './utils';
 
 export type ForkSpec = {
   name: string;
   network: string;
   deployment: string;
   blockNumber?: number;
   allocation?: number;
+  auxiliaryBase?: string;
 };
 
+export type Snapshot = {
+  snapshot: string;
+  auxiliarySnapshot?: string;
+}
+
 export class World {
-  hre: HardhatRuntimeEnvironment;
   base: ForkSpec;
   deploymentManager: DeploymentManager;
   snapshotDeploymentManager: DeploymentManager;
 
-  constructor(hre, base: ForkSpec) {
+  auxiliaryDeploymentManager?: DeploymentManager;
+  snapshotAuxiliaryDeploymentManager?: DeploymentManager;
+
+  constructor(base: ForkSpec) {
     // Q: should we really need to fork/snapshot the deployment manager?
-    this.hre = hre;
+    const hre = hreForBase(base);
     this.base = base;
     this.deploymentManager = new DeploymentManager(base.network, base.deployment, hre);
     this.snapshotDeploymentManager = this.deploymentManager;
+
+    if (this.base.auxiliaryBase) {
+      const auxiliaryBase = hre.config.scenario.bases.find(b => b.name === this.base.auxiliaryBase);
+      this.auxiliaryDeploymentManager = new DeploymentManager(auxiliaryBase.network, auxiliaryBase.deployment, hreForBase(auxiliaryBase));
+      this.snapshotAuxiliaryDeploymentManager = this.auxiliaryDeploymentManager;
+    }
   }
 
   isRemoteFork(): boolean {
     return this.base.network !== 'hardhat';
   }
 
-  async _snapshot(): Promise<string> {
+  async _snapshot(): Promise<Snapshot> {
     this.snapshotDeploymentManager = this.deploymentManager.fork();
-    return (await this.hre.network.provider.request({
+    const snapshot = await this.deploymentManager.hre.network.provider.request({
       method: 'evm_snapshot',
       params: [],
-    })) as string;
+    }) as string;
+    let auxiliarySnapshot: string;
+    if (this.auxiliaryDeploymentManager) {
+      this.snapshotAuxiliaryDeploymentManager = this.auxiliaryDeploymentManager.fork();
+      auxiliarySnapshot = await this.auxiliaryDeploymentManager.hre.network.provider.request({
+        method: 'evm_snapshot',
+        params: [],
+      }) as string;
+    }
```

Comment:
Oh hm 🤔

Reply:
which part are you thinking about?

Reply:
Just thinking that I see why the change I commented on above was made (worldSnapshot)